test/Succeed/Issue5683.agda

```diff
@@ -0,0 +1,17 @@
+
+open import Agda.Builtin.Equality
+open import Agda.Builtin.Sigma
+
+postulate
+  X  : Set
+  ∀≡ : ∀ (x : X) → x ≡ x
+
+variable x : X
+
+data C : Σ X (λ x → x ≡ x) → Set where
+  mkC :
+    let
+      eq : x ≡ x  -- don't generalize over x at eq
+      eq = ∀≡ x
```

Comment:
Why not `refl`?

Reply:
Why indeed? I copied the test case from the issue report.

Reply:
_Kopieren geht über studieren._

Reply:
My German is a bit non-existing, but I think that means "copying is superior to studying".

Reply:
Well, the original issue would not arise with `refl`, it's an artificial example to trigger the unexpected generalisation.

Reply:
I see, thanks!

Reply:
However, it can be simplified to `refl {x = x}` without the need of any postulates, I will submit a PR.